internal/transport/cart_handler.go

```diff
@@ -0,0 +1,191 @@
+package transport
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	domainErrs "innogocartapi/internal/domain_errs"
+	"innogocartapi/internal/models"
+	"innogocartapi/internal/services"
+	"log/slog"
+	"net/http"
+	"strconv"
+)
+
+type CartHandler struct {
+	cartService *services.CartService
+}
+
+func NewCartHandler(cartService *services.CartService) *CartHandler {
+	return &CartHandler{
+		cartService: cartService,
+	}
+}
+
+func (cartHandler *CartHandler) NewServeMux() *http.ServeMux {
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/carts", exceptionMiddleware(cartHandler.CartCreate))
+	mux.HandleFunc("/carts/{cartid}/items", exceptionMiddleware(cartHandler.AddToCart))
+	mux.HandleFunc("/carts/{cartid}/items/{itemid}", exceptionMiddleware(cartHandler.RemoveFromCart))
+	mux.HandleFunc("/carts/{cartid}", exceptionMiddleware(cartHandler.ViewCart))
```

Comment:
As I remember, this one should work out:
```suggestion
	mux.HandleFunc("POST /carts", exceptionMiddleware(cartHandler.CartCreate))
	mux.HandleFunc("POST /carts/{cartid}/items", exceptionMiddleware(cartHandler.AddToCart))
	mux.HandleFunc("DELETE /carts/{cartid}/items/{itemid}", exceptionMiddleware(cartHandler.RemoveFromCart))
	mux.HandleFunc("GET /carts/{cartid}", exceptionMiddleware(cartHandler.ViewCart))
```

Reply:
in this case you can get rid of manual checks in handlers